core/src/main/java/org/apache/iceberg/variants/Variants.java

```diff
@@ -85,6 +85,16 @@ public static ShreddedObject object(VariantMetadata metadata) {
     return new ShreddedObject(metadata);
   }
 
+  public static ShreddedObject object(VariantObject object) {
```

Comment:
Used to create a shredded object from an existing object when writing. It uses the object's metadata.

This avoids exposing `VariantObject.metadata` because metadata is carried by `Variant` instead of values.